res/controllers/Numark-Scratch-scripts.js

```diff
@@ -0,0 +1,417 @@
+    midi.sendShortMsg(0x98, 0x01, NumarkScratch.LOW_LIGHT);
+    midi.sendShortMsg(0x98, 0x02, NumarkScratch.LOW_LIGHT);
+    midi.sendShortMsg(0x99, 0x03, NumarkScratch.LOW_LIGHT);
+    midi.sendShortMsg(0x99, 0x04, NumarkScratch.LOW_LIGHT);
+    midi.sendShortMsg(0x99, 0x05, NumarkScratch.LOW_LIGHT);
+
+    // Send Serato SysEx messages to request initial state and unlock pads
+    midi.sendSysexMsg([0xF0, 0x00, 0x20, 0x7F, 0x00, 0xF7]);
+
+};
+
+NumarkScratch.shutdown = function() {
+    midi.sendSysexMsg([0xF0, 0x00, 0x20, 0x7F, 0x04, 0xF7]);
+};
+
+NumarkScratch.shift = function() {
+    NumarkScratch.shifted = true;
+    NumarkScratch.deck.shift();
+    NumarkScratch.effect.shift();
+};
+
+NumarkScratch.unshift = function() {
+    NumarkScratch.shifted = false;
+    NumarkScratch.deck.unshift();
+    NumarkScratch.effect.unshift();
+};
+
+NumarkScratch.EffectUnit = function(deckNumber) {
+
+    const inputUnshifted = function(channel, control, value, _status, _group) {
+        if (!this.isPress(channel, control, value)) {
+            return;
+        }
+        // Enable the current button
+        engine.setValue(this.group, this.inKey, true);
+
+        // Disable all other buttons in the same unit
+        NumarkScratch.effect[deckNumber - 1].effectButtons.forEach((effectButton) => {
+            if (effectButton !== this) {
+                engine.setValue(effectButton.group, effectButton.inKey, false);
+            }
+        });
+
+        // Disable all buttons in the other unit
+        NumarkScratch.effect[1 - (deckNumber - 1)].effectButtons.forEach((effectButton) => {
+            engine.setValue(effectButton.group, effectButton.inKey, false);
+        });
+    };
+
+    this.effectButtons = [];
+    for (let i = 0; i < 3; i++) {
+        this.effectButtons[i] = new components.Button({
+            group: `[EffectRack1_EffectUnit${deckNumber}_Effect${i + 1}]`,
+            midi: [0x98 + deckNumber - 1, (deckNumber - 1) * 3 + i],
+            inKey: "enabled",
+            outKey: "enabled", // Bind the LED state to the effect enabled state
+            shift: function() {
+                this.input = components.Button.prototype.input;
+                this.type = components.Button.prototype.types.toggle;
+            },
+            unshift: function() {
+                this.input = inputUnshifted;
+            },
+        });
+    }
+
+    this.paddle = new components.Button({
+        group: `[EffectRack1_EffectUnit${deckNumber}]`,
+        outKey: "enabled",
+        inKey: `group_[Channel${deckNumber}]_enable`,
+        input: function(channel, control, value, _status, _group) {
+            this.isPress = value !== 0;
+            engine.setValue("[EffectRack1_EffectUnit1]", `group_[Channel${deckNumber}]_enable`, this.isPress);
+            engine.setValue("[EffectRack1_EffectUnit2]", `group_[Channel${deckNumber}]_enable`, this.isPress);
+            this.trigger();
+        }
```

Comment:
what does the outKey here actually do?  The paddle doesn't have an LED as far as I can tell.

Reply:
yep, there's no need. done